includes/embeds/class-amp-base-embed-handler.php

```diff
@@ -160,6 +160,24 @@ protected function unwrap_p_element( DOMElement $node ) {
 		}
 	}
 
+	/**
+	 * Unwrap `<p>` element based on passed child tag name.
+	 *
+	 * @since 2.4.2
+	 *
+	 * @param Document $dom            Document.
+	 * @param string   $child_tag_name Child tag name.
+	 */
+	protected function unwrap_p_element_by_child_tag_name( $dom, $child_tag_name ) {
+		$nodes = $dom->xpath->query( "//p/{$child_tag_name}" );
+
+		if ( $nodes->length && $nodes instanceof DOMNodeList ) {
```

Comment:
Seems like the order is backwards, and I don't think `$nodes->length` needs to be checked since the `foreach` will do that:

```suggestion
		if ( $nodes instanceof DOMNodeList ) {
```